common/core/gpudevice.cpp

```diff
@@ -663,20 +668,32 @@ void GpuDevice::HandleRoutine() {
   // 2) Another app is having control of display and we
   //    we need to take control.
   // TODO: Add splash screen support.
-  lock_fd_ = open("/vendor/hwc.lock", O_RDONLY);
-  if (lock_fd_ == -1) {
+
+  if (flock(lock_fd_, LOCK_EX|LOCK_NB) != 0) {
     thread_sync_lock_.unlock();
-    return;
-  }
+    ITRACE("Another process is holding hwc lock, "
+           "wait until it releases the lock.");
 
-  thread_sync_lock_.unlock();
-  display_manager_->IgnoreUpdates();
+    display_manager_->IgnoreUpdates();
 
-  if (flock(lock_fd_, LOCK_EX) != 0)
-    ETRACE("Failed to wait on hwc lock.");
+    thread_sync_lock_.lock();
```

Comment:
Since in later code pieces a new lock will be issued, unlock here is to maintain the correct file locking status.

Removing relevant method calls on display_manager from lock region is to ensure no potential interleave locking among 2 classes. Underlying purpose is to avoid deadlock. If there is a chance that display_manager calls methods from class GpuDevice.